mitiq/lre/tests/test_lre.py

```diff
@@ -55,6 +71,29 @@ def test_lre_exp_value(degree, fold_multiplier):
     assert abs(lre_exp_val - ideal_val) <= abs(noisy_val - ideal_val)
 
 
+@pytest.mark.parametrize(
+    "degree, fold_multiplier, observable",
+    [(2, 2, obs_z), (2, 3, obs_z), (3, 4, obs_z)],
+)
+def test_lre_exp_value_with_observable(degree, fold_multiplier, observable):
+    """Verify LRE executors with observables work as expected."""
+    test_executor = Executor(mitiq_cirq.compute_density_matrix)
+    lre_exp_val = execute_with_lre(
+        test_cirq,
+        test_executor,
+        degree=degree,
+        fold_multiplier=fold_multiplier,
+        observable=observable,
+    )
+    assert abs(lre_exp_val - ideal_val) <= abs(noisy_val - ideal_val)
```

Comment:
```suggestion
```

I think the assertions you have below are great. This kind of assertion is great in theory, but in practice can lead to tests that fail randomly which is annoying.